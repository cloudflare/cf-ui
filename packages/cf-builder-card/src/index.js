import cardActions from './cardActions';
import CardActionTypes from './CardActionTypes';
import CardBuilder from './CardBuilder';
import cardReducer from './cardReducer';

export { cardActions, CardActionTypes, CardBuilder, cardReducer };
