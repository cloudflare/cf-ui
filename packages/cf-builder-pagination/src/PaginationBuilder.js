import React, { PropTypes } from 'react';
import { Pagination, PaginationItem } from 'cf-component-pagination';
import Icon from 'cf-component-icon';

class PaginationBuilder extends React.Component {
  render() {
    const {
      totalCount: totalItems,
      page,
      perPage,
      infoFormatter,
      loading,
      onPageChange
    } = this.props;

    const totalPages = Math.ceil(totalItems / perPage);
    const pageIndex = this.props.page - 1;
    const start = Math.max(pageIndex * perPage + 1, 1);
    const end = Math.min(start + perPage - 1, totalItems);

    const handleClickItem = page => {
      if (page >= 1 && page <= totalPages) {
        onPageChange(page);
      }
    };

    const items = [];

    let prevWasEllipsis = false;

    for (let i = 1; i <= totalPages; i++) {
      if (
        // First Page:
        i === 1 ||
        // Last Page:
        i === totalPages ||
        // Current Page:
        i === page ||
        // Within 2 of page:
        Math.abs(page - i) <= 2 ||
        // Within 3 of page and next to start or end: (Why? Because if we show an ellipsis instead of just a single number then wtf was the point)
        (Math.abs(page - i) <= 3 && (i === 2 || i === totalPages - 1))
      ) {
        prevWasEllipsis = false;
        items.push(
          <PaginationItem
            key={i}
            type={loading && page === i ? 'loading' : 'number'}
            label={'Page ' + i}
            active={page === i}
            onClick={() => handleClickItem(i)}
          >
            {i}
          </PaginationItem>
        );
      } else if (!prevWasEllipsis) {
        prevWasEllipsis = true;
        items.push(<PaginationItem key={i} type="ellipsis" />);
      }
    }

    const info = infoFormatter && infoFormatter(start, end, totalItems);

    return (
      <Pagination info={info}>
        <PaginationItem
          type="prev"
          label="Previous Page"
          disabled={page === 1}
          onClick={() => handleClickItem(page - 1)}
        >
          <Icon type="caret-left" label={false} />
        </PaginationItem>
        {items}
        <PaginationItem
          type="next"
          label="Next Page"
          disabled={page === totalPages}
          onClick={() => handleClickItem(page + 1)}
        >
          <Icon type="caret-right" label={false} />
        </PaginationItem>
      </Pagination>
    );
  }
}

PaginationBuilder.propTypes = {
  onPageChange: PropTypes.func.isRequired,
  loading: PropTypes.bool,

  totalCount: PropTypes.number.isRequired,
  page: PropTypes.number.isRequired,
  perPage: PropTypes.number.isRequired,

  infoFormatter: PropTypes.func
};

export default PaginationBuilder;
