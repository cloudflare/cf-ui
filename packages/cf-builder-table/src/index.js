import tableActions from './tableActions';
import TableActionTypes from './TableActionTypes';
import TableBuilder from './TableBuilder';
import TableBuilderPropTypes from './TableBuilderPropTypes';
import tableReducer from './tableReducer';

export {
  tableActions,
  TableActionTypes,
  TableBuilder,
  TableBuilderPropTypes,
  tableReducer
};
