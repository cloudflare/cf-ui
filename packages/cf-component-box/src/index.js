import { applyTheme } from 'cf-style-container';
import BoxUnstyled from './Box';
import BoxTheme from './BoxTheme';

const Box = applyTheme(BoxUnstyled, BoxTheme);

export { BoxUnstyled, Box, BoxTheme };
