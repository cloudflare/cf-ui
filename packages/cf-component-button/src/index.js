import Button from './Button';
import ButtonTheme from './ButtonTheme';
import ButtonGroup from './ButtonGroup';
import ButtonGroupTheme from './ButtonGroupTheme';

export { Button, ButtonTheme, ButtonGroup, ButtonGroupTheme };
