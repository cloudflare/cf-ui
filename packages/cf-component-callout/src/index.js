import Callout from './Callout';

export { Callout };
