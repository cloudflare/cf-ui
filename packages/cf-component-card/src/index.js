import { applyTheme } from 'cf-style-container';

import CardUnstyled from './Card';
import CardTheme from './CardTheme';
import CardBlockUnstyled from './CardBlock';
import CardBlockTheme from './CardBlockTheme';
import CardContentUnstyled from './CardContent';
import CardContentTheme from './CardContentTheme';
import CardControlUnstyled from './CardControl';
import CardControlTheme from './CardControlTheme';
import CardDrawersUnstyled from './CardDrawers';
import CardDrawersTheme from './CardDrawersTheme.js';
import CardMessagesUnstyled from './CardMessages';
import CardMessagesTheme from './CardMessagesTheme';
import CardSectionUnstyled from './CardSection';
import CardSectionTheme from './CardSectionTheme';
import CardTitleUnstyled from './CardTitle';
import CardTitleTheme from './CardTitleTheme';
import CardFooterUnstyled from './CardFooter';
import CardFooterTheme from './CardFooterTheme';
import CardToolbar from './CardToolbar';
import CardToolbarLink from './CardToolbarLink';
import CardPropTypes from './CardPropTypes';

const Card = applyTheme(CardUnstyled, CardTheme);
const CardBlock = applyTheme(CardBlockUnstyled, CardBlockTheme);
const CardContent = applyTheme(CardContentUnstyled, CardContentTheme);
const CardControl = applyTheme(CardControlUnstyled, CardControlTheme);
const CardDrawers = applyTheme(CardDrawersUnstyled, CardDrawersTheme);
const CardMessages = applyTheme(CardMessagesUnstyled, CardMessagesTheme);
const CardSection = applyTheme(CardSectionUnstyled, CardSectionTheme);
const CardTitle = applyTheme(CardTitleUnstyled, CardTitleTheme);
const CardFooter = applyTheme(CardFooterUnstyled, CardFooterTheme);

export {
  Card,
  CardUnstyled,
  CardTheme,
  CardContent,
  CardContentUnstyled,
  CardContentTheme,
  CardControl,
  CardControlUnstyled,
  CardControlTheme,
  CardDrawers,
  CardDrawersUnstyled,
  CardDrawersTheme,
  CardMessages,
  CardMessagesUnstyled,
  CardMessagesTheme,
  CardPropTypes,
  CardSection,
  CardSectionUnstyled,
  CardSectionTheme,
  CardBlock,
  CardBlockUnstyled,
  CardBlockTheme,
  CardTitle,
  CardTitleUnstyled,
  CardTitleTheme,
  CardFooter,
  CardFooterUnstyled,
  CardFooterTheme,
  CardToolbar,
  CardToolbarLink
};
