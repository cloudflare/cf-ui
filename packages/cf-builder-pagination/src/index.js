import PaginationBuilder from './PaginationBuilder';

export { PaginationBuilder };
