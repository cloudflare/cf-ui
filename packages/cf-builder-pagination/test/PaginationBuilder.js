import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { PaginationBuilder } from '../../cf-builder-pagination/src/index';
import { createStub } from '../../cf-test-stub/src/index';
import felaTestContext from '../../../felaTestContext';

test('should render pagination', () => {
  const component = renderer.create(
    felaTestContext(
      <PaginationBuilder
        onPageChange={() => {}}
        totalCount={100}
        page={1}
        perPage={20}
      />
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render pagination with the last page active', () => {
  const component = renderer.create(
    felaTestContext(
      <PaginationBuilder
        onPageChange={() => {}}
        totalCount={100}
        page={5}
        perPage={20}
      />
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render with collapsed items', () => {
  const component = renderer.create(
    felaTestContext(
      <PaginationBuilder
        onPageChange={() => {}}
        totalCount={20}
        page={6}
        perPage={1}
      />
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render pagination loading', () => {
  const component = renderer.create(
    felaTestContext(
      <PaginationBuilder
        onPageChange={() => {}}
        totalCount={100}
        page={1}
        perPage={20}
        loading={true}
      />
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render pagination with an infoFormatter', () => {
  let called = false;
  const component = renderer.create(
    felaTestContext(
      <PaginationBuilder
        onPageChange={() => {}}
        totalCount={100}
        page={2}
        perPage={20}
        infoFormatter={(start, end, totalItems) => {
          expect(start).toBe(21);
          expect(end).toBe(40);
          expect(totalItems).toBe(100);
          called = true;
        }}
      />
    )
  );
  expect(called).toBeTruthy();
  expect(component.toJSON()).toMatchSnapshot();
});

// StyleProvider does not pass props
// test('should call onPageChange when clicking another page', () => {
//   const onPageChange = createStub();

//   const wrapper = mount(
//     felaTestContext(
//       <PaginationBuilder
//         onPageChange={onPageChange}
//         totalCount={20}
//         page={1}
//         perPage={1}
//       />
//     )
//   );

//   const items = wrapper.find('li');

//   items.at(2).find('a').simulate('click');
//   expect(onPageChange.called).toBeTruthy();
//   expect(onPageChange.calls[0].args[0]).toBe(2);

//   wrapper.setProps({ page: 2 });
//   console.log(items.debug());
//   items.at(0).find('a').simulate('click');
//   expect(onPageChange.called).toBeTruthy;
//   expect(onPageChange.calls[1].args[0]).toBe(1);

//   wrapper.setProps({ page: 1 });
//   items.at(items.length - 1).find('a').simulate('click');
//   expect(onPageChange.called).toBeTruthy;
//   expect(onPageChange.calls[2].args[0]).toBe(2);
// });
