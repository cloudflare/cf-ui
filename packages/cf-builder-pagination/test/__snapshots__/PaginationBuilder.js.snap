// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render pagination 1`] = `
<div
  className="cf-1bek1ug"
  id={undefined}
  style={undefined}
>
  <ul
    aria-describedby={null}
    className="cf-123w5k0"
    role="navigation"
  >
    <li
      className="cf-tn889k"
      role={null}
    >
      <span
        aria-label="Previous Page"
        className="cf-3opov"
      >
        <i
          className="cf-icon cf-icon--caret-left"
          role={undefined}
        />
      </span>
    </li>
    <li
      className="cf-1j1psy1"
      role={null}
    >
      <span
        aria-label="Page 1"
        className="cf-3opov"
      >
        1
      </span>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 2"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 3"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        3
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 4"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        4
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 5"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        5
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Next Page"
        className="cf-1ex7c60"
        href="#"
        onClick={[Function]}
      >
        <i
          className="cf-icon cf-icon--caret-right"
          role={undefined}
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`should render pagination loading 1`] = `
<div
  className="cf-1bek1ug"
  id={undefined}
  style={undefined}
>
  <ul
    aria-describedby={null}
    className="cf-123w5k0"
    role="navigation"
  >
    <li
      className="cf-tn889k"
      role={null}
    >
      <span
        aria-label="Previous Page"
        className="cf-3opov"
      >
        <i
          className="cf-icon cf-icon--caret-left"
          role={undefined}
        />
      </span>
    </li>
    <li
      className="cf-1j1psy1"
      role={null}
    >
      <span
        aria-label="Page 1"
        className="cf-3opov"
      >
        <i
          className="cf-icon cf-icon--loading"
          role={undefined}
        />
      </span>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 2"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 3"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        3
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 4"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        4
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 5"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        5
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Next Page"
        className="cf-1ex7c60"
        href="#"
        onClick={[Function]}
      >
        <i
          className="cf-icon cf-icon--caret-right"
          role={undefined}
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`should render pagination with an infoFormatter 1`] = `
<div
  className="cf-1bek1ug"
  id={undefined}
  style={undefined}
>
  <ul
    aria-describedby={null}
    className="cf-123w5k0"
    role="navigation"
  >
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Previous Page"
        className="cf-1ex7c60"
        href="#"
        onClick={[Function]}
      >
        <i
          className="cf-icon cf-icon--caret-left"
          role={undefined}
        />
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 1"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="cf-1j1psy1"
      role={null}
    >
      <span
        aria-label="Page 2"
        className="cf-3opov"
      >
        2
      </span>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 3"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        3
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 4"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        4
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 5"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        5
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Next Page"
        className="cf-1ex7c60"
        href="#"
        onClick={[Function]}
      >
        <i
          className="cf-icon cf-icon--caret-right"
          role={undefined}
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`should render pagination with the last page active 1`] = `
<div
  className="cf-1bek1ug"
  id={undefined}
  style={undefined}
>
  <ul
    aria-describedby={null}
    className="cf-123w5k0"
    role="navigation"
  >
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Previous Page"
        className="cf-1ex7c60"
        href="#"
        onClick={[Function]}
      >
        <i
          className="cf-icon cf-icon--caret-left"
          role={undefined}
        />
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 1"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 2"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 3"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        3
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 4"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        4
      </a>
    </li>
    <li
      className="cf-1j1psy1"
      role={null}
    >
      <span
        aria-label="Page 5"
        className="cf-3opov"
      >
        5
      </span>
    </li>
    <li
      className="cf-tn889k"
      role={null}
    >
      <span
        aria-label="Next Page"
        className="cf-3opov"
      >
        <i
          className="cf-icon cf-icon--caret-right"
          role={undefined}
        />
      </span>
    </li>
  </ul>
</div>
`;

exports[`should render with collapsed items 1`] = `
<div
  className="cf-1bek1ug"
  id={undefined}
  style={undefined}
>
  <ul
    aria-describedby={null}
    className="cf-123w5k0"
    role="navigation"
  >
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Previous Page"
        className="cf-1ex7c60"
        href="#"
        onClick={[Function]}
      >
        <i
          className="cf-icon cf-icon--caret-left"
          role={undefined}
        />
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 1"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role="presentation"
    >
      <span
        aria-label="Page 2"
        className="cf-1ex7c60"
      >
        <span>
          …
        </span>
      </span>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 4"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        4
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 5"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        5
      </a>
    </li>
    <li
      className="cf-1j1psy1"
      role={null}
    >
      <span
        aria-label="Page 6"
        className="cf-3opov"
      >
        6
      </span>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 7"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        7
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 8"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        8
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role="presentation"
    >
      <span
        aria-label="Page 9"
        className="cf-1ex7c60"
      >
        <span>
          …
        </span>
      </span>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Page 20"
        className="cf-p16ik9"
        href="#"
        onClick={[Function]}
      >
        20
      </a>
    </li>
    <li
      className="cf-1hg1ctm"
      role={null}
    >
      <a
        aria-label="Next Page"
        className="cf-1ex7c60"
        href="#"
        onClick={[Function]}
      >
        <i
          className="cf-icon cf-icon--caret-right"
          role={undefined}
        />
      </a>
    </li>
  </ul>
</div>
`;
